compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/draganddrop/DragAndDrop.desktop.kt

```diff
@@ -54,7 +54,13 @@ actual class DragAndDropTransferData(
      */
     @property:ExperimentalComposeUiApi
     val onTransferCompleted: ((userAction: DragAndDropTransferAction?) -> Unit)? = null,
-)
+) {
+
+    init {
+        require(supportedActions.firstOrNull() != null) { "supportedActions may not be empty" }
```

Comment:
```suggestion
        require(supportedActions.firstOrNull() != null) { "supportedActions should not be empty" }
```

Reply:
"May" (or "must") in this context are actually better, as they imply a necessary requirement. "Should" has more of a suggestion connotation.

Reply:
Changed to `must`. `may` seems also is treated as not necessary?